components/TextFields/src/MDCTextInputBorderView.m

```diff
@@ -0,0 +1,171 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTextInputBorderView.h"
+
+static NSString *const MDCTextInputBorderViewBorderFillColorKey =
+    @"MDCTextInputBorderViewBorderFillColorKey";
+static NSString *const MDCTextInputBorderViewBorderPathKey =
+    @"MDCTextInputBorderViewBorderPathKey";
+static NSString *const MDCTextInputBorderViewBorderStrokeColorKey =
+    @"MDCTextInputBorderViewBorderStrokeColorKey";
+
+static inline NSString * _Nullable MDCNSStringFromCGLineCap(CGLineCap lineCap) {
+  NSString *lineCapString;
+  switch (lineCap) {
+    case kCGLineCapButt:
+      lineCapString = kCALineCapButt;
+      break;
+    case kCGLineCapRound:
+      lineCapString = kCALineCapRound;
+      break;
+    case kCGLineCapSquare:
+      lineCapString = kCALineCapSquare;
+      break;
+    default:
```

Comment:
Consider removing the default clause.

If we change this switch statement to:
switch(lineCap) {
  case kCGLineCapButt:
    return kCALineCapButt;
  case kCGLineCapRound:
    return kCALineCapROund;
  case kCGLineCapSquare:
    return kCALineCapSquare;
}
return nil;
}

If Apple adds another entry to CGLineCap in iOS 12.3, the compiler will automagically throw a warning along the lines of 'You aren't handling every case in your switch'.  Otherwise, with our current implementation someone will upgrade to iOS 12.3 and wonder why we aren't supporting kCGLineCapTriangle and we will have no warning.

Here and below.